### ISSUE TITLE
Jupyter Notebook tutorials

### ISSUE THREAD

**Issue by adamjstewart** [microsoft/torchgeo#38]
We need to figure out how to render Jupyter Notebooks in our documentation so that we can provide easy-to-use tutorials for new users. This should work similarly to https://pytorch.org/tutorials/.

Ideally I would like to be able to test these tutorials so that they stay up-to-date.

**Comment by Geethen** [comment]
Google Earth Engine, there are a few datasets available for ML as of now. 
[BigEarthNet](https://developers.google.com/earth-engine/datasets/catalog/TUBerlin_BigEarthNet_v1)
[LandCoverNet](https://samapriya.github.io/awesome-gee-community-datasets/projects/lcnet/)

I would be keen on assisting with this at some point

This also gave me an idea to contribute more datasets to the community catalog

**Comment by adamjstewart** [comment]
Does GEE support running jupyter notebooks? I've only ever used JavaScript in their code editor. It's hard to make any assumptions about data availability since the notebook needs to run on Colab, PC, and CI.

**Comment by Geethen** [comment]
Yes, it does via the GEE [Python API](https://developers.google.com/earth-engine/guides/python_install). 

Some drawbacks
1. The interactive leafmap/folium map will not stay alive. So you would have to opt for static images (that will need to be downloaded)
2. Perhaps more serious, is the user would need to download the data to their drive (or GD or GCS) which has been made easier with [geedim](https://github.com/dugalh/geedim) for image data (currently the workflow I use). However, I do not know of an instance where this can be avoided. I wonder if streaming the data as batches from GEE would be fast enough, or how much of a delay that will introduce.

**Comment by adamjstewart** [comment]
Okay, so this would be no different than our current approach of downloading data from Planetary Computer. Just another source of data.

**Comment by Geethen** [comment]
My apologies. I think it is going to be the case on all platforms for the foreseeable future (until GEE directly supports NNs-likely not any time soon).

Side note: in the geedim package the author used an approach based on rasterio to write image patches in chunks. Perhaps useful for inference

**Comment by adamjstewart** [comment]
Even if GEE directly supported NNs, they wouldn't support TorchGeo, so they aren't really relevant to us other than a possible data source. It would be much more fruitful to be able to directly support data in Colab or Planetary Computer. There's some work in progress on the PC side, but I'm not sure what's available in Colab.

**Comment by Geethen** [comment]
Good point. Yeah, that is likely going to be the case.

At the same time, once it comes to taking algorithms to data, paywalls and
restrictive quotas (Google colab (including pro) has become more
restrictive) will hinder, if not, prevent the geospatial community from
training models on global data.
For the sake of torchgeo, we may (at some point) get away with demos on
small datasets in something like colab + PC (hopefully sooner than later).

I havn't used PC much, but it felt like there is still huge ground left to
catch up to GEE (understandable given that GEE has had a ~10-year head
start- ). I am sure we all look forward to that day:)

Thanks for the exchange

**Comment by adamjstewart** [comment]
Yep, GEE used to have a lot more data, although I think PC might have already caught up in that front. GEE is still far more user friendly and easier to scale, so it's winning for non-CS people. But GEE is also very limited because it doesn't support NNs. In that sense, GEE is ~10 years behind TorchGeo 😄 

(the entire geospatial community is ~10 years behind the computer vision community, computer vision folks haven't used anything other than CNNs for over a decade)

We're hoping to provide something as easy as possible for geospatial researchers hoping to explore deep learning methods. Of course, TorchGeo isn't restricted to Colab or PC, you can use it on your laptop, supercomputer, or in the cloud (AWS, Azure, GCP, etc.). As long as you can get your hands on some data, and you can afford compute time, you can use TorchGeo.

**Comment by Geethen** [comment]
I don't think so. Have you seen all the datasets on GEE. Without the GEE
community catalog there are still way more datasets that are stored on GEE.
But, yes Google is slow in adding new datasets. I think even for CS people
(and non-geo people), user-friendliness is welcome. There is (limited)
support for NN on GEE, you could upload a trained NN model to GEE and then
run inference in GEE. I have also seen a paper where someone coded some NN
(from the ground up) in GEE.

TorchGeo is super useful for making it easier to build NNs (thanks for your
work). Building traditional tabular models is still often the way to go
because of how much easier it is. A lot of the time, the benefits of using
a DNN are not worth the effort of gathering the data+labels required for
these hungry models. My current mindset is to use an ensemble of both :)

There is one geospatial lab (Twitter handle: @xiaoxiang_zhu) I am aware of
that is at the cutting edge of applying computer vision (+NLP) stuff for
mapping.
